pkg/ebpf/dns_snooper.go

```diff
@@ -176,6 +169,36 @@ func (s *SocketFilterSnooper) pollStats() {
 	}
 }
 
+func (s *SocketFilterSnooper) start(filter *bpflib.SocketFilter) error {
+	setupErr := make(chan error)
+	s.wg.Add(1)
+	go func() {
+		defer s.wg.Done()
+
+		// Switch to root network namespace so we can snoop DNS traffic for all containers
+		runtime.LockOSThread()
+		defer runtime.UnlockOSThread()
+		rootNS, err := netns.GetFromPid(1)
```

Comment:
It looks like this reads from procfs, so PID 1 will be the system-probe, and not the non-namespace root PID

Reply:
yeah, sorry. I realized that testing it. I'm pushing a fix.